include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp

```diff
@@ -181,6 +196,8 @@ namespace alpaka
                 // After a block has been processed, the shared memory has to be deleted.
                 freeSharedVars(acc);
             }
+            // Restore the old schedule to not interfere with the rest
+            omp::setSchedule(oldSchedule);
```

Comment:
This should be called by a scope guard to guarantee that this function is called in case of exceptions. (like `freeSharedVars` throws)

Reply:
Good point, will fix